test/with-utility.connect.test.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import {
+  act,
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import useScript from 'react-script-hook';
+import { generateUseScriptMock } from './test-utils';
+import { withUtilityConnect } from '../src/with-utility-connect';
+
+jest.mock('react-script-hook', () => ({
+  __esModule: true,
+  default: jest.fn(),
+}));
+
+const generateSampleConfig = () => ({
+  env: 'sandbox',
+  client: 'Test Co.',
+  accessToken: 'this_is_a_super_secret_token',
+});
+
+class MockCredentialComponentClass extends React.Component {
+  render() {
+    const config = this.props.config ?? generateSampleConfig();
+    const { loading, error, open } = this.props.utilityConnect;
+    return (
+      <div>
+        {loading && <div>Loading...</div>}
+        {error && <div role="alert">{error.message}</div>}
+        <button type="button" disabled={loading} onClick={() => open(config)}>
+          {this.props.buttonLabel}
+        </button>
+      </div>
+    );
+  }
+}
```

Comment:
why do we want to mock this?

Reply:
There's no way to test a higher order component in React Testing Library without mocking a child component that takes it in. This mock handles the loading, open, and error props in easy-to-test ways.